datastacks/datastacks/pyspark/data_quality/utils.py

```diff
@@ -181,48 +184,31 @@ def publish_quality_results_table(
         ]
     )
 
-    emp_RDD = spark.sparkContext.emptyRDD()
-
-    data = spark.createDataFrame(data=emp_RDD, schema=dq_results_schema)
-
+    results_list = []
     for result in results:
-        try:
-            column_name = result.expectation_config.kwargs["column"]
-        except KeyError:
-            column_name = result.expectation_config.kwargs["column_A"]
+        column_name = result.expectation_config.kwargs.get("column", result.expectation_config.kwargs.get("column_A"))
         validator = result.expectation_config.expectation_type
-        try:
-            value_set = result.expectation_config.kwargs["value_set"]
-        except KeyError:
-            value_set = None
-        try:
-            threshold = result.expectation_config.kwargs["mostly"]
-        except KeyError:
-            threshold = None
-        unexpected_count = result.result["unexpected_count"]
-        unexpected_percent = result.result["unexpected_percent"]
-        success = result["success"]
-        row = spark.createDataFrame(
-            data=[
-                [
+        if not result.exception_info["raised_exception"]:
+            results_list.append(
+                (
                     data_quality_run_date,
                     datasource_name,
                     column_name,
                     validator,
-                    value_set,
-                    threshold,
-                    unexpected_count,
-                    unexpected_percent,
-                    success,
-                ]
-            ],
-            schema=dq_results_schema,
-        )
-        data = data.union(row)
-
-    data = data.coalesce(1)
-
-    save_dataframe_as_delta(spark, data, dq_path, overwrite=False, merge_keys=["data_quality_run_date"])
+                    result.expectation_config.kwargs.get("value_set"),
+                    result.expectation_config.kwargs.get("mostly"),
+                    result.result.get("unexpected_count"),
+                    result.result.get("unexpected_percent"),
+                    result.get("success"),
+                )
+            )
+        else:
+            logger.error(f"Failure while executing expectation {validator}, on column {column_name}.")
+            logger.error(f"Exception message: {result.exception_info['exception_message']}")
```

Comment:
This could be easy to miss if people are not looking at the logs of the spark jobs?
What if we instead still write out to the results table, but set the failure rate to 100% and list EXCEPTION RAISED or something like that? This way we could have monitoring and alerting flag further down the line that our configuration is invalid